include/quic/address.hpp

```diff
@@ -249,6 +250,37 @@ namespace oxen::quic
     template <>
     inline constexpr bool IsToStringFormattable<Address> = true;
 
+    struct RemoteAddress : public Address
+    {
+      private:
+        ustring remote_pubkey;
+
+      public:
+        RemoteAddress() = delete;
+
+        template <typename... Opt>
+        RemoteAddress(std::string_view remote_pk, Opt&&... opts) :
+                Address{std::forward<Opt>(opts)...}, remote_pubkey{to_usv(remote_pk)}
+        {}
+
+        template <typename... Opt>
+        RemoteAddress(ustring_view remote_pk, Opt&&... opts) : Address{std::forward<Opt>(opts)...}, remote_pubkey{remote_pk}
+        {}
+
+        ustring_view view_remote_key() const { return remote_pubkey; }
+        const ustring& get_remote_key() const { return remote_pubkey; }
+
+        RemoteAddress(const RemoteAddress& obj) : Address{*this}, remote_pubkey{obj.remote_pubkey} { _copy_internals(obj); }
+        RemoteAddress& operator=(const RemoteAddress& obj)
+        {
+            remote_pubkey = obj.remote_pubkey;
+            _copy_internals(obj);
```

Comment:
Here instead of `_copy_internals` it would be cleaner to invoke the parent classes' copy assignment operator, via:
```suggestion
            Address::operator=(obj);
```
to let Address worry about dealing with the Address internals.